plugins/modules/mso_schema_site_external_epg.py

```diff
@@ -160,7 +214,17 @@ def main():
             ops.append(dict(op="remove", path=op_path))
 
     elif state == "present":
-        l3out_dn = "uni/tn-{0}/out-{1}".format(tenant_name, l3out)
+        # Get external EPGs type from template level
+        external_epgs = [e.get("name") for e in schema_obj.get("templates")[template_idx]["externalEpgs"]]
```

Comment:
would it be option to get the external epg that matches the name instead of returning a list and indexing? This way I think you can simplify the code inside the if statement.